cron.php

```diff
@@ -0,0 +1,464 @@
+			add_action( 'wpccrij_confirm_scheduled_posts', array( $this, 'confirm_scheduled_posts' ) );
+		} else {
+			add_action( 'admin_notices', array( $this, 'admin_notice' ) );
+		}
+	}
+
+	/**
+	 * Set additional variables required for plugin functionality
+	 */
+	private function prepare() {
+		// Authentication
+		$this->secret = WP_CRON_CONTROL_SECRET;
+
+		// Internal jobs
+		$this->internal_jobs = array(
+			array(
+				'schedule' => 'wpccrij_minute',
+				'action'   => 'wpccrij_force_publish_missed_schedules',
+			),
+			array(
+				'schedule' => 'wpccrij_ten_minutes',
+				'action'   => 'wpccrij_confirm_scheduled_posts',
+			),
+		);
+
+		$this->internal_jobs_schedules = array(
+			'wpccrij_minute' => array(
+				'interval' => 1 * MINUTE_IN_SECONDS,
+				'display' => __( 'WP Cron Control Revisited internal job - every minute', 'wp-cron-control-revisited' ),
+			),
+			'wpccrij_ten_minutes' => array(
+				'interval' => 10 * MINUTE_IN_SECONDS,
+				'display' => __( 'WP Cron Control Revisited internal job - every 10 minutes', 'wp-cron-control-revisited' ),
+			),
+		);
+
+		// Prime lock cache if not present
+		wp_cache_add( $this->cache_key_lock, 0 );
+		wp_cache_add( $this->cache_key_lock_timestamp, time() );
+	}
+
+	/**
+	 * Block direct cron execution as early as possible
+	 */
+	public function block_direct_cron() {
+		if ( false !== strpos( $_SERVER['REQUEST_URI'], '/wp-cron.php' ) ) {
+			status_header( 403 );
+			wp_send_json_error( new WP_Error( 'forbidden', sprintf( __( 'Normal cron execution is blocked when the %s plugin is active.', 'wp-cron-control-revisited' ), 'WP-Cron Control Revisited' ) ) );
+		}
+	}
+
+	/**
+	 * Block the `spawn_cron()` function
+	 */
+	public function block_spawn_cron( $spawn_cron_args ) {
+		delete_transient( 'doing_cron' );
+
+		$spawn_cron_args['url']  = '';
+		$spawn_cron_args['key']  = '';
+		$spawn_cron_args['args'] = array();
+
+		return $spawn_cron_args;
+	}
+
+	/**
+	 * Register API routes
+	 */
+	public function rest_api_init() {
+		register_rest_route( $this->namespace, '/events/', array(
+			'methods'             => 'POST',
+			'callback'            => array( $this, 'get_events' ),
+			'permission_callback' => array( $this, 'check_secret' ),
+			'show_in_index'       => false,
+		) );
+
+		register_rest_route( $this->namespace, '/event/', array(
+			'methods'             => 'PUT',
+			'callback'            => array( $this, 'run_event' ),
+			'permission_callback' => array( $this, 'check_secret' ),
+			'show_in_index'       => false,
+		) );
+	}
+
+	/**
+	 * Include custom schedules used for internal jobs
+	 */
+	public function register_internal_events_schedules( $schedules ) {
+		return array_merge( $schedules, $this->internal_jobs_schedules );
+	}
+
+	/**
+	 * Schedule internal jobs
+	 */
+	public function schedule_internal_events() {
+		$when = strtotime( sprintf( '+%d seconds', 2 * $this->job_queue_window_in_seconds ) );
```

Comment:
I honestly have no idea; I might've been using one of Core's time constants before switching to the class property. Regardless, I'll remove it; shifting by the job queue window just avoids scheduling something that'll immediately miss its schedule.